joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForStatementsCreator.scala

```diff
@@ -4,26 +4,34 @@ import io.joern.gosrc2cpg.parser.ParserAst._
 import io.joern.gosrc2cpg.parser.{ParserKeys, ParserNodeInfo}
 import io.joern.gosrc2cpg.utils.Operator
 import io.joern.x2cpg.Ast
-import io.shiftleft.codepropertygraph.generated.{DispatchTypes, Operators}
+import io.shiftleft.codepropertygraph.generated.{ControlStructureTypes, DispatchTypes, Operators}
 
 import scala.util.Try
 
 trait AstForStatementsCreator { this: AstCreator =>
-  def astForBlockStatement(blockStmt: ParserNodeInfo): Seq[Ast] = {
-
-    val newBlockNode = blockNode(blockStmt)
+  def astForBlockStatement(blockStmt: ParserNodeInfo, order: Int = -1): Ast = {
+    val newBlockNode = blockNode(blockStmt, Defines.empty, Defines.voidTypeName).order(order).argumentIndex(order)
     scope.pushNewScope(newBlockNode)
+    var currOrder = 1
```

Comment:
Instead of using `currOrder`, use the `withIndex` method exposed by `X2Cpg`'s `AstCreatorBase`. This is similar to `zipWithIndex` but will start at 1.

Reply:
@DavidBakerEffendi We cannot use `withIndex` here as we are updating `currOrder` based on Ast length from inside Expression